### PR TITLE
Fix duplicated nodes when re adding package paths

### PR DESCRIPTION
**Pull request by pinzart90**
https://jira.autodesk.com/browse/DYN-3852

Prevent nodes from being duplicated if the package paths are re-added to the node and package paths list.

### CODE REVIEW

File: src/DynamoPackages/PackageLoader.cs [hunk 1]
```diff
@@ -297,6 +294,7 @@ private void TryLoadPackageIntoLibrary(Package package)
 
                 package.LoadState.SetAsLoaded();
                 PackgeLoaded?.Invoke(package);
+                PackagesLoaded?.Invoke(loadedAssemblies);
```
Comment (aparajit-pratap) [inline src/DynamoPackages/PackageLoader.cs:297]: Can you explain how moving this call into `TryLoadPackageIntoLibrary` fixes the bug?

Comment (mjkkirschner) [reply src/DynamoPackages/PackageLoader.cs:297]: agree - I'm not clear on how this works.

Comment (pinzart90) [reply src/DynamoPackages/PackageLoader.cs:297]: There are conditions in the `TryLoadPackageIntoLibrary` that (if evaluated as true) will skip loading the package.
Ex: If the package was already loaded or if it was marked as unloaded, or if an exception is thrown...

In the case where we do not load a package then we should not invoke the `PackagesLoaded` event for that package.

In the unit test I wrote (for example) the `PackagesLoaded` event is not called if the package was already loaded

**Review by sm6srw — APPROVED** [review]
LGTM!

**Review by aparajit-pratap — APPROVED** [review]

**Pull request merged**